File: product-service/data/mapper/product.go
package mapper

import (
	"github.com/fredoliveira-ca/products-golang-java/product-service/app/web/dto"
	"github.com/fredoliveira-ca/products-golang-java/product-service/domain"
)

// Of is ...
func Of(product domain.Product) dto.Product {
	return dto.Product{
		ID:           product.ID,
		PriceInCents: product.PriceInCents,
		Description:  product.Description,
		Title:        product.Title,
	}
}

File: discount-service/src/main/java/com/fredoliveira/discountcalculator/domain/Product.java
package com.fredoliveira.discountcalculator.domain;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class Product {

  private final String id;
  private final String title;
  private final String description;
  private final Long priceInCents;
  private final Discount discount;

}
